cmd/opm/alpha/veneer/basic.go

```diff
@@ -21,10 +21,19 @@ func newBasicVeneerRenderCmd() *cobra.Command {
 	)
 	cmd := &cobra.Command{
 		Use:   "basic basic-veneer-file",
-		Short: "Generate a declarative config blob from a single 'basic veneer' file",
-		Long:  `Generate a declarative config blob from a single 'basic veneer' file, typified as a declarative configuration file where olm.bundle objects have no properties`,
-		Args:  cobra.ExactArgs(1),
+		Short: "Generate a declarative config blob from a single 'basic veneer' file \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Long:  "Generate a declarative config blob from a single 'basic veneer' file, typified as a declarative configuration file where olm.bundle objects have no properties \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Args:  cobra.MaximumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
+			// Handle different input argument types
+			// When no arguments or "-" is passed to the command,
+			// assume input is coming from stdin
+			// Otherwise open the file passed to the command
+			data, source, err := OpenFileOrReadStdin(cmd, args)
+			if err != nil {
+				log.Fatalf("unable to read %q: %v", source, err)
```

Comment:
couple of nits here:
1. we're not reading stdin yet, we're just opening it for reading.
2. do we need `OpenFileOrReadStdin` to be consumed outside this package? If not, make it unexported.

```suggestion
			data, source, err := OpenFileOrStdin(cmd, args)
			if err != nil {
				log.Fatalf("unable to open %q: %v", source, err)
```